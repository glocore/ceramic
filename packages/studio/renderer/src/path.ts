import { useSystemStore } from "./store";

export const getPrintablePath = (path: string) => {
  const homeDir = useSystemStore.getState().homeDir;

  if (!homeDir) {
    return path;
  }

  const printablePath = path.replace(homeDir, "~");

  return printablePath;
};
