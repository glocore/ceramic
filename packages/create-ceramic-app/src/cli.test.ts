import runTest, { DOWN, ENTER } from "cli-prompts-test";
import fs from "node:fs";
import path from "node:path";
import { afterEach, beforeEach, describe, expect, test } from "vitest";

describe.sequential("cli", () => {
  const testRoot = path.join(__dirname, "temp");

  beforeEach(() => {
    fs.mkdirSync(testRoot, { recursive: true });
  });

  afterEach(() => {
    fs.rmSync(testRoot, { recursive: true, force: true });
  });

  describe("no project name supplied", () => {
    test("scaffolds project without input", async () => {
      const { exitCode, stdout } = await runTest(
        [path.join(__dirname, "cli.ts")],
        [ENTER],
        { testPath: testRoot }
      );
      expect(exitCode).toBe(0);
      expect(stdout).toContain("Ceramic app created at my-ceramic-app");
    });

    test("scaffolds project with project name as argument", async () => {
      const projectName = "test-app";

      const { exitCode, stdout } = await runTest(
        [path.join(__dirname, "cli.ts")],
        [`${projectName}${ENTER}`],
        { testPath: testRoot }
      );
      expect(exitCode).toBe(0);
      expect(stdout).toContain(`Ceramic app created at ${projectName}`);
    });
  });

  describe("with project name supplied", () => {
    test("scaffolds project with project name", async () => {
      const projectName = "test-app";

      const { exitCode, stdout } = await runTest(
        [path.join(__dirname, "cli.ts"), projectName],
        [ENTER],
        { testPath: testRoot }
      );
      expect(exitCode).toBe(0);
      expect(stdout).toContain(`Ceramic app created at ${projectName}`);
    });

    test("scaffolds project without current directory name", async () => {
      const { exitCode, stdout } = await runTest(
        [path.join(__dirname, "cli.ts"), "."],
        [ENTER],
        { testPath: testRoot }
      );
      expect(exitCode).toBe(0);
      expect(stdout).toContain("Ceramic app created at temp");
    });
  });

  describe("non-empty directory", () => {
    const createTestFile = (destination?: string) => {
      const fileName = "test.txt";
      fs.writeFileSync(
        path.join(destination ?? "./", fileName),
        "test",
        "utf-8"
      );
      return fileName;
    };

    test("cancels operation", async () => {
      createTestFile(testRoot);

      const { exitCode, stdout } = await runTest(
        [path.join(__dirname, "cli.ts"), "."],
        [ENTER],
        { testPath: testRoot }
      );

      expect(exitCode).toBe(0);
      expect(stdout).toContain("Operation cancelled.");
    });

    test("removes existing files and continues", async () => {
      const testFileName = createTestFile(testRoot);

      const { exitCode, stdout } = await runTest(
        [path.join(__dirname, "cli.ts"), "."],
        [DOWN, ENTER],
        { testPath: testRoot }
      );

      expect(exitCode).toBe(0);
      expect(stdout).toContain("Ceramic app created at temp");
      expect(fs.readdirSync(testRoot)).not.toContain(testFileName);
    });
  });

  test("generates expected files", async () => {
    const { exitCode } = await runTest(
      [path.join(__dirname, "cli.ts"), "."],
      [ENTER],
      { testPath: testRoot }
    );

    expect(exitCode).toBe(0);

    const files = (
      fs.readdirSync(testRoot, { recursive: true }) as string[]
    ).filter((filePath) =>
      fs.lstatSync(path.join(testRoot, filePath)).isFile()
    );

    expect(files).toMatchSnapshot();

    for (const filePath of files) {
      const fileContents = fs
        .readFileSync(path.join(testRoot, filePath), "utf-8")
        .toString();

      await expect(fileContents).toMatchFileSnapshot(
        `./__snapshots__/tempate/${filePath}.snap`
      );
    }
  });

  test("adds the project name in the relevant places", async () => {
    const projectName = "test-app";

    const { exitCode } = await runTest(
      [path.join(__dirname, "cli.ts"), projectName],
      [ENTER],
      { testPath: testRoot }
    );

    expect(exitCode).toBe(0);

    expect(
      fs
        .readFileSync(path.join(testRoot, projectName, "README.md"), "utf-8")
        .toString()
    ).toContain(projectName);

    const pkg = JSON.parse(
      fs
        .readFileSync(path.join(testRoot, projectName, "package.json"), "utf-8")
        .toString()
    );

    expect(pkg.name).toEqual(projectName);
  });
});
