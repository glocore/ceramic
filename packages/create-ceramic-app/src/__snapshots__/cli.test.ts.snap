// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cli > generates expected files 1`] = `
[
  ".gitignore",
  ".prettierignore",
  "README.md",
  "app.config.ts",
  "package.json",
  "postcss.config.mjs",
  "tailwind.config.mjs",
  "tsconfig.json",
  ".vscode/settings.json",
  "public/android-chrome-192x192.png",
  "public/android-chrome-512x512.png",
  "public/apple-touch-icon.png",
  "public/favicon-16x16.png",
  "public/favicon-32x32.png",
  "public/favicon.ico",
  "public/favicon.png",
  "public/site.webmanifest",
  "src/api.ts",
  "src/client.tsx",
  "src/global-middleware.ts",
  "src/routeTree.gen.ts",
  "src/router.tsx",
  "src/ssr.tsx",
  "src/components/DefaultCatchBoundary.tsx",
  "src/components/NotFound.tsx",
  "src/components/PostError.tsx",
  "src/components/UserError.tsx",
  "src/routes/__root.tsx",
  "src/routes/_pathlessLayout.tsx",
  "src/routes/deferred.tsx",
  "src/routes/index.tsx",
  "src/routes/posts.$postId.tsx",
  "src/routes/posts.index.tsx",
  "src/routes/posts.route.tsx",
  "src/routes/posts_.$postId.deep.tsx",
  "src/routes/redirect.tsx",
  "src/routes/users.$userId.tsx",
  "src/routes/users.index.tsx",
  "src/routes/users.route.tsx",
  "src/styles/app.css",
  "src/utils/loggingMiddleware.tsx",
  "src/utils/posts.tsx",
  "src/utils/seo.ts",
  "src/utils/users.tsx",
  "src/routes/_pathlessLayout/_nested-layout.tsx",
  "src/routes/api/users.$id.ts",
  "src/routes/api/users.ts",
  "src/routes/_pathlessLayout/_nested-layout/route-a.tsx",
  "src/routes/_pathlessLayout/_nested-layout/route-b.tsx",
]
`;
